src/main/java/uk/gov/hmcts/reform/pcqloader/helper/PayloadValidationHelper.java

```diff
@@ -49,10 +50,14 @@ public void validateDisabilityImpact(PcqAnswers answers) {
     }
 
     public void validateLanguageLevel(PcqAnswers answers) {
-        //If user has selected the main language as English/Welsh or Prefer Not to Say then
-        //set the language level to null.
-        if (answers.getEnglishLanguageLevel() != null && answers.getLanguageMain() != null
-            && (answers.getLanguageMain() == 1 || answers.getLanguageMain() == 0)) {
+        //If user has selected the main language as English, Welsh, English or Welsh (legacy language option)
+        // or Prefer Not to Say
+        // then set the language level to null.
+        boolean isLanguageSelected = Optional.ofNullable(answers.getLanguageMain())
+            .map(languageMain -> List.of(0, 1, 3, 4).contains(languageMain))
+            .orElse(false);
```

Comment:
I would advise inverse the logic, that should make it clearer what is being done here. E.g. instead of having `boolean isLanguageSelected`, change that to `boolean nonEnglishLanguageSelected`. And instead of checking whether option is one of 0, 1, 3, 4 - check if the option is 2.
With those changes, comment could be adjusted to make it a lot clearer - 
`// if "other" language, meaning non-English/Welsh is selected, keep the english level, else set it to null as it is irrelevant for native speaker.`